fineract-provider/src/main/java/org/apache/fineract/infrastructure/event/external/jobs/SendAsynchronousEventsTasklet.java

```diff
@@ -0,0 +1,97 @@
+ * under the License.
+ */
+package org.apache.fineract.infrastructure.event.external.jobs;
+
+import java.sql.Types;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.fineract.avro.MessageV1;
+import org.apache.fineract.infrastructure.core.config.FineractProperties;
+import org.apache.fineract.infrastructure.core.service.DateUtils;
+import org.apache.fineract.infrastructure.core.service.RoutingDataSourceServiceFactory;
+import org.apache.fineract.infrastructure.core.service.database.DatabaseSpecificSQLGenerator;
+import org.apache.fineract.infrastructure.event.external.producer.ExternalEventProducer;
+import org.apache.fineract.infrastructure.event.external.repository.ExternalEventRepository;
+import org.apache.fineract.infrastructure.event.external.repository.domain.ExternalEvent;
+import org.apache.fineract.infrastructure.event.external.repository.domain.ExternalEventStatus;
+import org.apache.fineract.infrastructure.event.external.serializer.ExternalEventDataSerializer;
+import org.springframework.batch.core.StepContribution;
+import org.springframework.batch.core.scope.context.ChunkContext;
+import org.springframework.batch.core.step.tasklet.Tasklet;
+import org.springframework.batch.repeat.RepeatStatus;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@RequiredArgsConstructor
+@Component
+public class SendAsynchronousEventsTasklet implements Tasklet {
+
+    private final RoutingDataSourceServiceFactory dataSourceServiceFactory;
+    private final FineractProperties fineractProperties;
+    private final ExternalEventRepository repository;
+    private final ExternalEventProducer eventProducer;
+    private final ExternalEventDataSerializer eventDataSerializer;
+    private final DatabaseSpecificSQLGenerator sqlGenerator;
+
+    @Override
+    public RepeatStatus execute(StepContribution contribution, ChunkContext chunkContext) {
+
+        try {
+            int readBatchSize = getBatchSize();
+            List<Long> externalEventIds = getTobeSentEventsBatch(readBatchSize);
+            sendEventAndUpdateEventSentStatus(externalEventIds);
+        } catch (Exception e) {
+            log.info(e.getMessage());
+        }
+        return RepeatStatus.FINISHED;
+    }
+
+    private void sendEventAndUpdateEventSentStatus(List<Long> externalEventIds) throws Exception {
+        for (Long eventId : externalEventIds) {
+            ExternalEvent event = repository.findById(eventId).get();
+            MessageV1 eventData = eventDataSerializer.serialize(event);
+            eventProducer.sendEvent(eventData);
+            event.setStatus(ExternalEventStatus.SENT);
+            event.setSentAt(DateUtils.getOffsetDateTimeOfTenant());
+            repository.save(event);
+
+        }
+    }
+
+    private List<Long> getTobeSentEventsBatch(int readBatchSize) {
```

Comment:
No need to pass the batchsize as a param here, you could call it from within the method.